mnist_lenet/__init__.py

```diff
@@ -83,20 +84,21 @@ def __init__(self, conv_activations, num_channels, image_shape,
         self.top_mlp_dims = top_mlp_dims
         self.border_mode = border_mode
 
-        parameters = zip(conv_activations, filter_sizes, feature_maps,
-                         pooling_sizes)
+        conv_parameters = zip(conv_activations, filter_sizes, feature_maps)
 
         # Construct convolutional layers with corresponding parameters
-        self.layers = [ConvolutionalLayer(filter_size=filter_size,
-                                          num_filters=num_filter,
-                                          pooling_size=pooling_size,
-                                          activation=activation.apply,
-                                          conv_step=self.conv_step,
-                                          border_mode=self.border_mode,
-                                          name='conv_pool_{}'.format(i))
-                       for i, (activation, filter_size, num_filter,
-                               pooling_size)
-                       in enumerate(parameters)]
+        self.layers = list(interleave([
```

Comment:
I merged too quick and then realized, that I don't really like the idea of using `interleave` in our basic examples. We cater here to a broad audience, 99% of which do not use `toolz`.

Can you please do me a favour and use simpler Python here?